### ISSUE TITLE
Gradle 4.10.3 throws initializationError with Java 11

### ISSUE THREAD

**Issue by izeye** [gradle/gradle#8058]
### Expected Behavior

Gradle 4.10.3 works with Java 11 with the following test source (`src/test/java`):

```
package com.izeye.test.jdk;

import org.junit.Test;

public class ApplicationTests {

	@Test
	public void test() {
	}

	class Something {
	}

}
```

### Current Behavior

It fails with the following errors:

```
> Task :test FAILED

com.izeye.test.jdk.ApplicationTests$Something > initializationError FAILED
    java.lang.Exception

com.izeye.test.jdk.ApplicationTests$Something > initializationError FAILED
    java.lang.Exception

com.izeye.test.jdk.ApplicationTests$Something > initializationError FAILED
    java.lang.Exception

com.izeye.test.jdk.ApplicationTests$Something > initializationError FAILED
    java.lang.Exception

5 tests completed, 4 failed
```

### Context

N/A

### Steps to Reproduce (for bugs)

1. Clone this sample project: https://github.com/izeye/jdk-tests
2. Run the following command with JDK 11:

```
./gradlew clean check
```

I confirmed it's working with Gradle 5.0 with the sample which is on a different branch (`gradle-5.0`): https://github.com/izeye/jdk-tests/tree/gradle-5.0

### Your Environment

 * Build scan URL: 

N/A

**Comment by ljacomet** [comment]
Full Java 11 support is only available for Gradle 5.0 and above.

**Comment by izeye** [comment]
@ljacomet Ah, sorry for the noise. I should have checked it first.
